### ISSUE TITLE
example / tests, taking truth value of a symbol

### ISSUE THREAD

**Issue by Martin-S-Weber** [ctford/traversy#5]
https://github.com/ctford/traversy/blob/master/test/traversy/test/lens.cljc#L44

`(is (#(and (= % [2 3 4]) vector?) (-> [1 2 3] (update each inc))))`

in other words, is `(#(and (= % [2 3 4]) vector?) (-> [1 2 3] (update each inc)))` truthy
in other words, is `((fn [arg] (and (= arg [2 3 4]) vector?)) (-> [1 2 3] (update each inc)))` truthy
in other words, is both `(= arg [2 3 4])` and `vector?` truthy
in other words, is the _application_ of `=` on both `arg` and `[2 3 4]` truthy, as well as the _symbol_ `vector?` (which it better is in a clojure?)

same applies next line for seq?

I assume you meant something like

`(is (#(and (= % [2 3 4]) (vector? %)) (-> [1 2 3] (update each inc))))` ?

**Comment by Martin-S-Weber** [comment]
for L45, this should make the test fail btw:

```
user=> (-> [1 2 3] (t/update t/each inc))
[2 3 4]
user=> (-> [1 2 3] seq (t/update t/each inc))
[2 3 4]
user=> (#(and (= % [2 3 4]) seq?) (-> [1 2 3] seq (t/update t/each inc)))
#object[clojure.core$seq_QMARK___4132 0x3a58c388 "clojure.core$seq_QMARK___4132@3a58c388"]
user=> (#(and (= % [2 3 4]) (seq? %)) (-> [1 2 3] seq (t/update t/each inc))))
false
```

**Comment by ctford** [comment]
Oh, that does look like a mistake. Thanks!

**Comment by Martin-S-Weber** [comment]
the following line has the same problem and more crucially makes your test fail...
this isn't fully fixed yet.
